eden/converter/graph/gspan.py

```diff
@@ -28,37 +24,67 @@ def gspan_to_eden(input=None, options=dict()):
         yield gspan_to_networkx(header, string_list)
 
 
-def gspan_to_networkx(header, string_list):
-    graph = nx.Graph()
-    graph.graph['id'] = header
-    for line in string_list:
-        if line.strip():
-            line_list = line.split()
-            fc = line_list[0]
-            # process vertices
-            if fc in ['v', 'V']:
-                vid = int(line_list[1])
-                vlabel = line_list[2]
-                # lowercase v indicates active viewpoint
-                if fc == 'v':
-                    weight = 1
-                else:  # uppercase v indicates no-viewpoint, in the new EDeN this is simulated via a smaller weight
-                    weight = 0.1
-                graph.add_node(vid, ID=vid, label=vlabel, weight=weight)
-                # extract the rest of the line  as a JSON string that contains all attributes
-                attribute_str = ' '.join(line_list[3:])
-                if attribute_str.strip():
-                    attribute_dict = json.loads(attribute_str)
-                    graph.node[vid].update(attribute_dict)
-            # process edges
-            if fc == 'e':
-                srcid = int(line_list[1])
-                destid = int(line_list[2])
-                elabel = line_list[3]
-                graph.add_edge(srcid, destid, label=elabel)
-                attribute_str = ' '.join(line_list[4:])
-                if attribute_str.strip():
-                    attribute_dict = json.loads(attribute_str)
-                    graph.edge[srcid][destid].update(attribute_dict)
+def gspan_to_networkx(header, lines):
+    """Take a string list in the extended gSpan format and returns a NetworkX graph.
+
+    Keyword arguments:
+    header -- string to be used as id for the graph
+    lines  -- string list in extended gSpan format 
+    """
+    # remove empty lines
+    lines = [line for line in lines if line.strip()]
```

Comment:
It's redundant ... you are iterating over the same object 2 times and if a graph is big this will end up entirely in memory.